pkg/utils/openshift/utils.go

```diff
@@ -48,6 +51,69 @@ func MapKnownVersion(info platform.PlatformInfo) platform.OpenShiftVersion {
 		"1.10+": "3.10",
 		"1.11+": "3.11",
 		"1.13+": "4.1",
+		"1.14+": "4.2",
+		"1.16+": "4.3",
 	}
 	return platform.OpenShiftVersion{Version: k8sToOcpMap[info.K8SVersion]}
 }
+
+/*
+Supported OpenShift platform
+Supported version format : Major.Minor.Patch
+	e.g.: 2.3.4
+ */
+func CompareOpenShiftVersion(cfg *rest.Config, version string) (int, error) {
+	isOcp, err := IsOpenShift(cfg)
+	if err != nil {
+		return -1, err
+	}
+	if !isOcp {
+		return -1, errors.New("There is no OpenShift platform detected.")
+	}
+	info, err := GetPlatformInfo(cfg)
+	if err != nil {
+		return -1, err
+	}
+	curVersion := MapKnownVersion(info)
+	return CompareVersion(curVersion.Version, version)
```

Comment:
We can use LookupOpenShiftVersion here instead and prevent the need to utilize the version mapping and converting between so many types. Requires some tweaks to the compare func (see comment below). Something here like:
```
ocVersion, err := LookupOpenShiftVersion(cfg)
if err != nil {
  return -1, err
}
return CompareVersion(ocVersion, OpenShiftVersion{ Version: version })
```

Reply:
The LookupOpoenshiftVersion tests are like the following:
4.1.x : no permission is needed, return correct version
4.2.x: permission is needed, return correct version
4.3.x: no exception tells need any permission, but no version info returned here either w/ or w/o putting the permission info in csv.

It seems failing on 4.3 platform. I am still investigating. But welcome any idea.

Reply:
What 'permission info' are you putting in the CSV for previous versions? Can you include a snippet here of what you're adding? It's possible we need to revisit the way we get that info altogether if it requires users to escalate permissions for simple lookup.

Reply:
screen pasted.

Reply:
I finished up the ask I had on the other project I'm working with, so I'll set aside some hours in the morning to deep-dive this and figure out next step. I'd also like to understand why 4.3 is behaving differently... if they've changed the API endpoint again, it's just not reliable enough to count on IMO.

Reply:
ver | GetPlatformInfo() ending'+' is not guaranteed | OpenAPISchema() | clusterversion resource query:return correct ver
-- | -- | -- | --
4.1 | 1.13 | v1.13.4 | permission needed
4.2 | 1.14+ | v1.14.6 | permission needed
4.3 | 1.16+ | v0.0.0 | permission needed
3.11 | 1.11+ | v3.11.69 | failed - resource doesn't exist



Tried to break the methods down to individual API calls to make things a bit clear.

Other operator uses either clusterversion or clusteroperator resources, but it needs permission.
OpenAPI call doesn't seem to work in 4.3.
Though we have to maintain a mapping, GetPlatformInfo() appears to be a better choice for a library to have user friendliness and accuracy.

Reply:
Have you tried this approach? I wanted to wholesale plug this in as the new version check code and try it out, but have not had the chance. I don't believe it requires any permissions, but I do not know what all versions it supports (it does support 4.3 as it was written for that specifically) 

https://github.com/atlasmap/atlasmap-operator/blob/master/pkg/util/utils.go#L39-L68

Reply:
Permission is needed, they are actually calling the clusterversion resource like we do.